pkg/controller/controller.go

```diff
@@ -334,6 +343,36 @@ func ttls() []string {
 	return []string{"15m", "30m", "1h", "2h", "4h"}
 }
 
-func categories() []string {
-	return []string{"explanation"}
+func (c *Controller) categories() []string {
```

Comment:
I'm confused about these functions:

1. Why do we need them?
2. If we do need them, can we pre-compute these values once during controller instantiation instead of doing this iteration each time? Right now, this will be called on each request, which feels rather inefficient.
3. What data/structure is actually required? We can pass maps and slices into the Go templates directly and extract keys/values in the template engine.

Reply:
Currently categories() is used for two things:

1. set the default value of the drop down (relevant code [here](https://github.com/abcxyz/jvs/blob/main/pkg/controller/controller.go#L125))
2. validate the form and show corresponding errors (relevant code [here](https://github.com/abcxyz/jvs/blob/main/pkg/controller/controller.go#L258))

Existing code is just a hard-code string array, so it is ok to call them on every request. Good point that we should do the calculation beforehand. I have adjusted the code to calculate it when the controller is constructed.

`What data/structure is actually required? We can pass maps and slices into the Go templates directly and extract keys/values in the template engine.` <- I don't think we can pass the map and slices for the two use cases of `categories`. `getFormDetails` is doing something similar and it is the caller of `categoryPairs`

Reply:
https://github.com/abcxyz/jvs/pull/303

Reply:
Yes, the logic is much simpler after merging in #303 . Currently I directly add more entries to the categories map, let me know what you think